src/repositories/getPosts.js

```diff
@@ -0,0 +1,35 @@
+import { connectionDb} from "../database/db.js";
+
+export const getPostsUser= async(res, userId, followStatus)=>{
+    try{
+        const statusFollow = followStatus;
+        let response;
+        
+        if(statusFollow =="ok"){
+            
+            response = (await connectionDb.query(
+                `SELECT users.username, users."pictureUrl", posts.*, 
+                COALESCE(COUNT(likes."postId"),0) AS "numberOfLikes"
+                 FROM posts 
+                 LEFT JOIN users ON posts."userId" = users.id
+                 LEFT JOIN likes ON likes."postId" = posts.id
```

Comment:
```suggestion
                 LEFT JOIN likes ON likes."postId" = posts.id
                 (select count (*) from likes where "postId"=posts.id and "userId"=$1)::int as liked
```
pode adicionar essa linha antes de dar o merge

Reply:
Posso sim, pode deixar!